### PR TITLE
PACA-786: RDA pipeline job smoke tests

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[PACA-786](https://jira.cms.gov/browse/PACA-786)

**User Story or Bug Summary:**

Create specific smoke tests for both the FISS/MCS RDA API jobs to verify that on deployment they are still able to communicate with the RDA API. This will allow us to cancel a deployment if problem appears in the latest build.

---

### What Does This PR Do?

Builds on the framework established in PACA-785 to define specific smoke tests for the FISS and MCS RDA API pipeline jobs.  Leverages the existing `RdaSource` and `RdaSink` implementations to allow a single implementation of the tests for both pipelines.

The smoke tests for both jobs perform the same set of checks (just on different gRPC calls):

- read from the database to ensure database connectivity is present
- call RDA API version RPC to verify API connectivity is present
- retrieve 3 claim objects from the RDA API to verify claims can be downloaded and transformed

Note: Only the `StandardGrpcRdaSource` requires a smoke test.  Adding one to `DLQGrpcRdaSource` would be redundant since it uses the same configuration settings and library methods.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:
* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* [PACA-785](https://github.com/CMSgov/beneficiary-fhir-data/pull/1420)

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by KyRobbins — DISMISSED** [review]
Not sure where it would go, but wondering if there should be a comment somewhere stating that there is no implemented smoke test for the DLQGrpcRdaSource because it utilizes the exact same resources as the StandardGrpcRdaSource

**Review by KyRobbins — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/StandardGrpcRdaSource.java [hunk 1]
```diff
@@ -103,6 +103,52 @@ public StandardGrpcRdaSource(
...
+   */
+  @Override
+  public boolean performSmokeTest(RdaSink<TMessage, TClaim> sink) throws Exception {
+    boolean successful = false;
+    log.info("smoke test: starting");
+
+    // Query the database to get a starting sequence number.  This verifies that the database is
+    // accessible.
+    final long startingSequenceNumber =
+        sink.readMaxExistingSequenceNumber().orElse(MIN_SEQUENCE_NUM);
```
Comment (ericbuckley) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/StandardGrpcRdaSource.java:124]: If we start from the rda_api_progress table, there's a chance that when we deploy there is no more data for us to consume and a connection to the RDA API would just be waiting for more data to come in.  Am I reasoning this correctly?

Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/StandardGrpcRdaSource.java:124]: Yes.  If we pass the value from the `rda_api_progress` table to the RDA API when we start downloading claims then we might have to wait if there are no newer claims available.  That's why we start from`MIN_SEQUENCE_NUM` when we actually start retrieving claims in the next step.

Comment (ericbuckley) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/StandardGrpcRdaSource.java:124]: oh I see, its just for logging, my mistake

**Review by ericbuckley — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**